### PR TITLE
updated with modification

### PR DESCRIPTION
**Pull request by selroc**
Separated the lines so that cpu is always set.

### CODE REVIEW

**Comment by selroc** [comment]
I will add another line, if device is not set at that point the user has not specified -device or -d, which should default to the first gpu, which is 0.

**Comment by selroc** [comment]
> Not sure this is the good way, maybe this other one is better:
> if (s == "") {device=0;} else {device=stoi(s);}

This should be:
if (s.empty()) {device=0;} else {device=stoi(s);}

it should default to gpu 0 if the user specifies only -d without argument. But I'm not sure, probably you want to throw an error there.

File: Args.cpp [hunk 1]
```diff
@@ -107,6 +107,7 @@ void Args::parse(string line) {
     else if (key == "-cpu") { cpu = s; }
     else if (key == "-time") { timeKernels = true; }
     else if (key == "-device" || key == "-d") { device = stoi(s); }
+    if (cpu.empty()) { cpu = getHwName(device) + "-" + getBoardName(device) + "-" + str(device); }
```
Comment (valeriob01) [inline Args.cpp]: here if the user does not specify -d the variable 'device' will not contain the gpu number, to avoid this I am proposing this change:

Comment (valeriob01) [reply Args.cpp]: else if (key == "-device" || key == "-d") { device = stoi(s);
    if (cpu.empty()) { cpu = getHwName(device) + "-" + getBoardName(device) + "-" + str(device); }
}

This way only if -d is specified we assign cpu name.

**Comment by preda** [comment]
A similar change has been applied, closing this, thanks!

**Pull request closed without merging**